### ISSUE TITLE
toBeArrayOfSize() throwing error on constructing message

### ISSUE THREAD

**Issue by robross0606** [JamieMason/expect-more#35]
When using:
```javascript
expect(arrayOfObjects).toBeArrayOfSize(n)
```
I am getting errors from expect-more on creation of the jest error message when the test fails:
```
  TypeError: Cannot convert object to primitive value
      at Array.join (<anonymous>)
      at Array.toString (<anonymous>)
      at message (node_modules/expect-more-jest/dist/to-be-array-of-size.js:7:51)
      at message (node_modules/expect-more-jest/dist/lib/create-result.js:6:62)
      at getMessage (node_modules/expect/build/index.js:209:15)
      at processResult (node_modules/expect/build/index.js:327:25)
      at Object.toBeArrayOfSize (node_modules/expect/build/index.js:396:16)
```

**Comment by JamieMason** [comment]
Thanks @robross0606 I'll take a look

**Comment by JamieMason** [comment]
I've taken a look and so far can't reproduce your error in jest 24.9.0 or 25.2.3. 

Could you clone and edit `packages/expect-more-jest/test/to-be-array-of-size.spec.ts` to add a test that reproduces the behaviour?

```
git clone https://github.com/JamieMason/expect-more.git
cd expect-more
yarn install
yarn build
yarn test --watch --no-coverage packages/expect-more-jest/test/to-be-array-of-size.spec.ts
```

I can pick up from there.

**Comment by robross0606** [comment]
The reason your tests aren't working is that you're not trying them on _Objects_.  Only primitives from what I can see in the repo.  And the test has to _fail_ in order to cause the problem.

```javascript
it('provides expect().toBeArrayOfSize()', () => {
  expect([{ value: 'i' }, { value: 'contain' }, { value: 4 }, { value: 'objects' }]).toBeArrayOfSize(5);
});
```

**Comment by robross0606** [comment]
Hmmm... I just tried that on your test system and it worked okay.

Let me see what's different about my environment (other than not using Typescript).

**Comment by JamieMason** [comment]
Thanks Rob, I tried something very similar to this although I only used empty objects which I doubt would matter 🤔. Yeah if you could list jest, node etc versions we can maybe narrow it down that way.

**Comment by robross0606** [comment]
jest v25.2.6
node v12.14.1
npm v6.14.3

**Comment by JamieMason** [comment]
Thanks Rob I'll take a look tomorrow.

**Comment by robross0606** [comment]
Now I can't even reproduce it in my own code.  😕

**Comment by robross0606** [comment]
I was able to hit an extremely similar error again:
![image](https://user-images.githubusercontent.com/2965467/80824488-875ee380-8bac-11ea-80fc-6ed01bda6ae8.png)
In this case, `value` was an array instead of an `object`.  This caused the test to fail, but generation of the failure message caused this error.  The debugger result of `value` are:
![image](https://user-images.githubusercontent.com/2965467/80824593-c3924400-8bac-11ea-88ef-7b955b38504f.png)

**Comment by robross0606** [comment]
For some reason the runtime is having trouble converting `value` to a string in 
```javascript
"expected " + value + " to be an array containing exactly " + size + " items"
```

**Comment by robross0606** [comment]
From what I'm reading, your use of `value` in this expression assumes that all Object in `value` are created with a prototype and hence have a `toString()` function.  In this case, the results are coming back from a GraphQL call and apparently have no prototype so conversion of `value` to a string for the message fails. This is also why it was hard to reproduce as a test since using literal syntax would always have prototypes.  I was able to reproduce this bug using this:
```javascript
require('expect-more-jest')

describe('Bugs', () => {
  const prototypeLess = []

  beforeAll(() => {
    for (let i = 0; i < 5; i++) {
      prototypeLess[i] = createObjectWithoutPrototype(i + 1)
    }
  })

  it('Exception on failing test with array of objects', typeName => {
    expect(prototypeLess).toBeArrayOfSize(1)
  })
})

function createObjectWithoutPrototype(index) {
  const obj = Object.create(null)
  obj.name = `Object ${index}`
  obj.id = index
  return obj
}
```

**Comment by JamieMason** [comment]
Brilliant info @robross0606, thank you. I will take a look this weekend.

**Comment by robross0606** [comment]
Here is a more formal test you could use:
```javascript
describe('Prototypeless Bug', () => {
  const withPrototypes = []
  const withoutPrototypes = []

  beforeAll(() => {
    const createTestObject = (index, withProto) => {
      const obj = withProto ? {} : Object.create(null)
      obj.name = `Object ${index}`
      obj.id = index
      return obj
    }
    for (let i = 0; i < 5; i++) {
      withPrototypes[i] = createTestObject(i + 1, true)
      withoutPrototypes[i] = createTestObject(i + 1, false)
    }
  })

  it('Exception on failing test with array of objects', typeName => {
    let expectedError
    try {
      expect(withPrototypes).toBeArrayOfSize(1)
    } catch (error) {
      expect(error).toMatchInlineSnapshot(
        `[Error: expected [object Object],[object Object],[object Object],[object Object],[object Object] to be an array containing exactly 1 items]`
      )
      expectedError = error
    }
    expect(expect(withoutPrototypes).toBeArrayOfSize(1)).toThrow(error.message)
  })
})
```
This proves that object literals create objects with prototypes since the first set of expectations work.  The second usage should match behavior of the first but fails due to objects not having prototypes.

**Comment by JamieMason** [comment]
Released in 4.0.2, please let me know of any issues and thanks for all your work on investigating this issue.
